Update ADX guide to tutorial format (#1387)
astronomer/docs: learn/airflow-azure-data-explorer.md

---

> title: "Run an Azure Data Explorer Query with Airflow"

Reviewer: ```suggestion
title: "Run an Azure Data Explorer query with Airflow"
```

sidebar_label: "Azure Data Explorer"
description: "Learn how to orchestrate Azure Data Explorer queries with your Apache Airflow DAGs."
id: airflow-azure-data-explorer

> tags: ["Integrations", "Azure", "DAGs"]

Reviewer: ```suggestion
```

- The [Astro CLI](https://docs.astronomer.io/astro/cli/get-started).

## Step 1: Configure your ADX cluster to work with Airflow

To allow Airflow to communicate with your ADX database, you need to configure service principal authentication. To do this, create and register an Azure AD service principal, then give that principal permission to access your ADX database. See [Create an Azure Active Directory application registration in Azure Data Explorer](https://docs.microsoft.com/en-us/azure/data-explorer/provision-azure-ad-app) for more details.

## Step 2: Populate your ADX database

Populate your ADX database with demo data. See [Quickstart: Ingest sample data into ADX](https://learn.microsoft.com/en-us/azure/data-explorer/ingest-sample-data?tabs=ingestion-wizard) for instructions on ingesting the `StormEvents` sample dataset. If you are working with an existing ADX cluster that already contains data, you can skip this step.

## Step 3: Configure your Astro project

> Now that you have your Azure resources configured, you can move on to Airflow setup. Using the Astro CLI:

Reviewer: ```suggestion
Now that you have your Azure resources configured, you can move on to setting up Airflow.
```
